src/coreclr/jit/codegenlinear.cpp

```diff
@@ -625,6 +626,11 @@ void CodeGen::genCodeForBBlist()
                 switch (block->GetJumpKind())
                 {
                     case BBJ_ALWAYS:
+                        // We might skip generating the jump via a peephole optimization.
+                        // If that happens, make sure a NOP is emitted as the last instruction in the block.
+                        emitNop = true;
+                        break;
+
```

Comment:
I added an `assert(false)` to that case in the old code to see if I could get it to hit during a SuperPMI replay, and it never hit across all collections. Also in the new code, I added an assert that `BBJ_ALWAYS` has a jump before trying to emit the jump, so that we never have a `BBJ_ALWAYS` that "falls into" nothing at the end of the block list -- that also never hit.